### PR TITLE
Lambda sg update

### PR DESCRIPTION
**Pull request by ttheune**

### CODE REVIEW

**Comment by ttheune** [comment]
Merge conflict with changes made for assume role, I will clean those up on approval.

File: docker/foxpass_sg_update_tool.py [hunk 1]
```diff
@@ -73,6 +75,21 @@ def main():
     update_security_group(add_ports, clear_ports, rules, security_group, task_name)
 
 
+# In Lambda run this instead of main()
+def lambda_handler(event, context):
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    ec2 = boto3.resource('ec2')
+    ecs = boto3.client('ecs')
+    cluster, sg_id, task_name = (os.environ.get('CLUSTER'), os.environ.get('SG_ID'), os.environ.get('TASK_NAME'))
```
Comment (aren) [inline docker/foxpass_sg_update_tool.py]: make this three separate lines

File: docker/foxpass_sg_update_tool.py [hunk 2]
```diff
@@ -73,6 +75,21 @@ def main():
     update_security_group(add_ports, clear_ports, rules, security_group, task_name)
 
 
+# In Lambda run this instead of main()
+def lambda_handler(event, context):
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
```
Comment (aren) [inline docker/foxpass_sg_update_tool.py:81]: normally these two lines go at the top of the file, after the imports.

File: docker/foxpass_sg_update_tool.py [hunk 3]
```diff
@@ -73,6 +75,21 @@ def main():
...
+    ec2 = boto3.resource('ec2')
+    ecs = boto3.client('ecs')
+    cluster, sg_id, task_name = (os.environ.get('CLUSTER'), os.environ.get('SG_ID'), os.environ.get('TASK_NAME'))
+    security_group = ec2.SecurityGroup(sg_id)
+    active_ports = get_ports(cluster, ecs, task_name)
+    rules = get_current_sg_rules(security_group, task_name)
+    add_ports, clear_ports = ports_to_change(active_ports, rules)
+    update_security_group(add_ports, clear_ports, rules, security_group, task_name)
+    logger.info('Active: {}\tAdded: {}\tCleared: {}'.format(active_ports, add_ports, clear_ports))
+
```
Comment (aren) [inline docker/foxpass_sg_update_tool.py:93]: if this function throws an exception, does lambda catch it and show it to us somewhere?

**Review by aren — CHANGES_REQUESTED** [review]

File: docker/foxpass_sg_update_tool.py (hunk 2 above)
Comment (ttheune) [reply docker/foxpass_sg_update_tool.py:81]: Left in lambda_handler() as that's the only place it's currently used.  Will move out once logger goes to wider ultilization

File: docker/foxpass_sg_update_tool.py (hunk 3 above)
Comment (ttheune) [reply docker/foxpass_sg_update_tool.py:93]: Yes

**Review by aren — APPROVED** [review]

**Comment by aren** [comment]
After speaking with Travis, this is approved in its current form.

**Pull request merged**